### PR TITLE
MakeMake - PanelSidetray

### PR DESCRIPTION
**Pull request by dChiamp**
Components included in this PR are:

1. PanelSidetray

## Notes
Has separate story for active item. 
Added a click event handler that may not be needed
Is this used on mobile?

### CODE REVIEW

File: src/components/PanelSidetray/PanelSidetray.vue [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            justify-content: center;
+            align-items: center;
+
+            h4 {
+                color: var(--color-company);
+                font-size: 16px;
+                font-weight: 300;
+                opacity: 0;
+                transition: opacity 0.2s ease-in-out;
+            }
```
Comment (drewbaker) [inline src/components/PanelSidetray/PanelSidetray.vue]: Don't need to nest these. Give H4 a class!

Also H4 will need some padding so it positions nicely in the thumbnail

File: src/components/PanelSidetray/PanelSidetray.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
+<template lang="html">
+    <div class="panel-sidetray">
+        <nuxt-link
+            v-for="(item, i) in items"
+            v-if="item.featuredImage.sourceUrl"
+            :key="item.id"
+            :to="item.uri"
+            @click.native="handleClick"
```
Comment (drewbaker) [inline src/components/PanelSidetray/PanelSidetray.vue]: Don't need this click event. I'm going to make any click on the entire component close the panel.

**Review by drewbaker — CHANGES_REQUESTED** [review]
Can you add the little film icon into this. Position in off the left outside of the panel (position: absolute; left: -20px (or whatever).

Screen Shot 2020-07-15 at 4 22 19 PM

The last thumbnail shouldn't be hard up against the bottom of the thumbtray. Should have same spacing around it as the top.

**Comment by dChiamp** [comment]
OK Made these updates.

I don't think the `icon-reel` svg should be part of this component but I included it anyway.
I would think it's its own position fixed component on the page that translates the entire page over to reveal `panel-sidetray` on click:
Screen Shot 2020-07-15 at 5 05 43 PM

I repositioned things to make that easier.
Also I had to remove  `overflow-y: scroll`  to show the svg (see note in component).

I know we shouldnt use the `index` /  `i` as a key, so you'll notice there's a duplicate key error in the console bc of how i mocked up manyPages in the story.  It wont be an issue with real data and i didnt want to use `index` so i left it intentionally.

File: src/components/PanelSidetray/PanelSidetray.vue [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            >
+                <div class="scrim">
+                    <h4 v-if="i == activeIndex" class="title">
+                        Currently Viewing
+                    </h4>
+                    <h4 v-else class="title" v-html="item.title" />
+                </div>
+            </wp-image>
+        </nuxt-link>
+    </div>
```
Comment (drewbaker) [inline src/components/PanelSidetray/PanelSidetray.vue]: How you do this, and have it work correctly is this:

```
<div class="panel-sidetray"> // height --unit-100vh, position: fixed, right: 0; transform: translateX(100%);
      <svg-film> // Position: absolute, -100px;
       
      <div class="items"> // Height 100%, overflow-y: auto;
              <nuxt-link/>
              <nuxt-link/>
              <nuxt-link/>
      </div>
</div>
```

Getting both the thumbtray and the film SVG to animate smoothly in sync is difficult, but having them together like this makes it really easy.

Have a style in it for the open state, like this:

```
.panel-sidetray {

      .sidetray-opened & {
             transform: translate(0);
      }
}
```

I will then toggle that style on body (so I can also dim the page etc).

**Review by drewbaker — CHANGES_REQUESTED** [review]
See my notes for how to do this correctly

**Pull request merged**